tests/core/test_filter_pred.py

```diff
@@ -0,0 +1,88 @@
+import asyncio
+
+import pytest
+
+from conftest import (
+    generate_unique_name,
+)
+from lahja import (
+    BaseEvent,
+    ConnectionConfig,
+    Endpoint,
+)
+
+
+@pytest.mark.asyncio
+async def test_subscribed_events(event_loop: asyncio.AbstractEventLoop) -> None:
+    class TestSubscriptionEvent(BaseEvent):
+        pass
+
+    endpoint = Endpoint()
+    endpoint.subscribe(TestSubscriptionEvent, lambda event: None)
+
+    assert endpoint.subscribed_events == {TestSubscriptionEvent}
+
+    class TestStreamEvent(BaseEvent):
+        pass
+
+    async def stream() -> None:
+        async for _ in endpoint.stream(TestStreamEvent):  # noqa: F841
+            pass
+
+    future = asyncio.ensure_future(stream())
+    await asyncio.sleep(0)  # give stream a chance to be scheduled
+
+    assert endpoint.subscribed_events == {TestSubscriptionEvent, TestStreamEvent}
+
+    future.cancel()
+    endpoint.stop()
+
+
+class TestEvent(BaseEvent):
```

Comment:
Kind of annoying, but if you name this `EventForTesting` then it pytest won't try to collect it as a test class.

Reply:
I think we're save, pytest doesn't seem to be collecting this: 

```
(lahja) brian@rhythm:~/ef/lahja$ pytest --collect-only
========================================================================================================= test session starts =========================================================================================================
platform linux -- Python 3.7.0, pytest-4.0.2, py-1.8.0, pluggy-0.9.0 -- /home/brian/.local/share/virtualenvs/lahja-2m5rLbjb/bin/python3.7
cachedir: .pytest_cache
rootdir: /home/brian/ef/lahja, inifile: pytest.ini
plugins: xdist-1.25.0, forked-1.0.2, asyncio-0.9.0
collected 22 items                                                                                                                                                                                                                    
<Module 'tests/core/test_basics.py'>
  <Function 'test_request'>
  <Function 'test_request_can_get_cancelled'>
  <Function 'test_response_must_match'>
  <Function 'test_stream_with_break'>
  <Function 'test_stream_with_num_events'>
  <Function 'test_stream_can_get_cancelled'>
  <Function 'test_stream_cancels_when_parent_task_is_cancelled'>
  <Function 'test_wait_for'>
  <Function 'test_wait_for_can_get_cancelled'>
<Module 'tests/core/test_broadcast_config.py'>
  <Function 'test_broadcasts_to_all_endpoints'>
  <Function 'test_broadcasts_to_specific_endpoint'>
<Module 'tests/core/test_connect.py'>
  <Function 'test_can_not_connect_conflicting_names_blocking'>
  <Function 'test_can_not_connect_conflicting_names'>
  <Function 'test_rejects_duplicates_when_connecting_blocking'>
  <Function 'test_rejects_duplicates_when_connecting'>
  <Function 'test_rejects_duplicates_when_connecting_nowait'>
<Module 'tests/core/test_filter_pred.py'>
  <Function 'test_subscribed_events'>
  <Function 'test_remote_subscribed_events'>
  <Function 'test_can_wait_for_changes'>
<Module 'tests/core/test_import.py'>
  <Function 'test_import'>
<Module 'tests/core/test_internal.py'>
  <Function 'test_internal_propagation'>
<Module 'tests/core/test_stop.py'>
  <Function 'test_can_stop'>
```